### ISSUE TITLE
Annotation sharing: should you see the layer config of the annotation owner?

### ISSUE THREAD

**Issue by fm3** [scalableminds/webknossos#5080]
When sharing your annotation it may be desirable that the other user sees it just as you do, i.e. with your layer settings (same visibility + intensity settings). Let’s discuss if this is a good idea and when to copy settings from where to where

**Comment by philippotto** [comment]
I think, the most plausible way would be to provide a checkbox in the sharing modal à la: "Use my current view settings (e.g., layer visibility, intensity)". Then, the settings are persisted somehow so that they can be restored when the URL is opened. However, it's probably hard to pinpoint the exact "when" of the snapshot creation. Especially, when we want to remove the explicit save button from the modal.

That's why an easier variant could be to just copy the settings from the owner when the link is opened (as you probably meant in the issue description). This would probably be "good enough" for most usecases, anyway. However, we would need to pay attention that the wording is precise in the UI so that the user understands that their actions *after* creating the sharing link can influence the behavior of the link.

The layer settings are not version-managed by any chance? Then, the solution could be to allow both variants..

**Comment by fm3** [comment]
> The layer settings are not version-managed by any chance?

Not at this time, never had a reason to. I’d think the overhead is also not really justified for this use case

> copy the settings from the owner when the link is opened

Yes, that could be a solution.

Another problem concerns the state of the viewing user’s settings. Assuming they are a logged-in user, and not just a one-time viewer. The annotation author’s settings for both the dataset and annotation tab are copied. Where are they copied to? Are they just used while viewing this particular annotation? What happens if the viewing user changes a setting afterwards? Is that persisted to their own view configuration? I’m afraid both doing so and not doing so could create confusion. I don’t know how to clearly communicate this. (Also I’m not 100% clear about the settings behavior myself, the infos from both tabs are saved in the same place, right? UserDatasetViewConfiguration?)

**Comment by philippotto** [comment]
We should probably discuss this in person :speech_balloon: Additional relevant topics could be:

- sharing the activated mapping
- sharing loaded isosurfaces
- more?

**Comment by normanrz** [comment]
Could we encode the layer settings in the url? Maybe with an integrated backend-implemented URL shortening.

**Comment by philippotto** [comment]
Most of the discussion above is outdated by now. The current sharing code can encode layer-specific things in the URL and already does that for mappings, meshes, connectomes etc.

It's probably a good idea to at least add the following:
- layer visibility

For other properties, such as layer intensity, I'm not too sure if we want to do this by default, as it would overwrite the preferences of a user. Options I see:
- let the sharer decide whether these properties should be included
- let the user with whom the link was shared decide after opening if they want to apply "all" settings.

**Comment by fm3** [comment]
I agree that layer visibility should be added. The decisions on the latter two points should not block that
